core/test/base/array.cpp

```diff
@@ -284,6 +284,17 @@ TYPED_TEST(Array, CanBeResized)
 }
 
 
+TYPED_TEST(Array, VewCanBeResetNotResized)
+{
+    TypeParam data[] = {1, 2, 3};
+    auto view = gko::Array<TypeParam>::view(this->exec, 3, data);
+    view.resize_and_reset(1);
+
+    EXPECT_EQ(view.get_const_data(), nullptr);
+    ASSERT_EQ(view.get_num_elems(), 1);
```

Comment:
nullptr strongly suggests that the view is empty, should it have non-zero size then?

Reply:
Well yes, that depends if we do something or not with this size argument to the reset as you state earlie. For now I left it as is, but indeed I was considering always setting to 0 in this case. I guess another possibility is indeed to make this view into an array of 1 element. I'd like to have some feedback on what is the most useful here in fact.

Reply:
So far, I decided to stick with the current model, but I'm making it clear that whenever the data is just a `nullptr`, then `num_elems_` is always 0. I think if you want to change your view to an array, there is always the move operation to do that now so there's no need to add this functionality. It also helps keeping the border between the two clear: it's crossed only thanks to the move operator=, and I think that's easy enough to understand and reason with.

Reply:
An example where the `nullptr` behavior could cause issues:
If we have a CSR matrix that is built on top of `Array::view` col_idxs/values, using `apply` to compute the SpGEMM product with this matrix as output would segfault/fail: `csr_mtx->apply(csr_mtx2, csr_view_mtx)`, as the SpGEMM internally has to resize the arrays.